bahmnicore-api/src/main/java/org/bahmni/module/bahmnicore/dao/impl/PatientDaoImpl.java

```diff
@@ -44,219 +42,216 @@
-        if (!shouldMatchExactPatientId) {
-            String partialIdentifier = "%" + patientIdentifier;
-            Query querytoGetPatients = sessionFactory.getCurrentSession().createQuery(
-                    "select pi.patient " +
-                            " from PatientIdentifier pi " +
-                            " where pi.identifier like :partialIdentifier ");
-            querytoGetPatients.setString("partialIdentifier", partialIdentifier);
-            return querytoGetPatients.list();
-        }
-
-        Patient patient = getPatient(patientIdentifier);
-        List<Patient> result = (patient == null ? new ArrayList<Patient>() : Arrays.asList(patient));
-        return result;
-    }
-
-    @Override
-    public List<RelationshipType> getByAIsToB(String aIsToB) {
-        Query querytoGetPatients = sessionFactory.getCurrentSession().createQuery(
-                "select rt " +
-                        " from RelationshipType rt " +
-                        " where rt.aIsToB = :aIsToB ");
-        querytoGetPatients.setString("aIsToB", aIsToB);
-        return querytoGetPatients.list();
-    }
+	private SessionFactory sessionFactory;
+
+	private Internationalizer i18n;
+
+	@Autowired
+	public PatientDaoImpl(SessionFactory sessionFactory, Internationalizer i18n) {
+		this.sessionFactory = sessionFactory;
+		this.i18n = i18n;
+	}
+
+	@Override
+	public List<PatientResponse> getPatients(String identifier, String name, String customAttribute,
+			String addressFieldName, String addressFieldValue, Integer length,
+			Integer offset, String[] customAttributeFields, String programAttributeFieldValue,
+			String programAttributeFieldName, String[] addressSearchResultFields,
+			String[] patientSearchResultFields, String loginLocationUuid, Boolean filterPatientsByLocation, Boolean filterOnAllIdentifiers) {
+
+		validateSearchParams(customAttributeFields, programAttributeFieldName, addressFieldName);
+
+		ProgramAttributeType programAttributeType = getProgramAttributeType(programAttributeFieldName);
+
+		PatientSearchBuilder builder = new PatientSearchBuilder(sessionFactory)
+				.withPatientName(name)
+				.withPatientIdentifier(identifier, filterOnAllIdentifiers)
+				.withPatientAttributes(customAttribute, getPersonAttributeIds(customAttributeFields), getPersonAttributeIds(patientSearchResultFields))
+				.withProgramAttributes(programAttributeFieldValue, programAttributeType)
+				.withLocation(loginLocationUuid, filterPatientsByLocation);
+
+		if (!i18n.isEnabled() || StringUtils.isEmpty(addressFieldValue)) {
+			builder.withPatientAddress(addressFieldName, addressFieldValue, addressSearchResultFields);
+		}
+		else {
+			// when i18n is enabled the address is in a list of matched address i18n codes, if any
+			List<String> codedAddressFieldValues = i18n.getAddressMessageKeysByLikeName(addressFieldValue);
+			builder.withPatientAddressInList(addressFieldName, codedAddressFieldValues, addressSearchResultFields);
+		}
```

Comment:
@angshu, there was an issue when `addressFieldValue` was empty (or null) and I18N was enabled. Here is a last min fix. Basically when `addressFieldValue` is empty it goes to the default route.